pcsx2/gui/AppMain.cpp

```diff
@@ -529,11 +531,34 @@ extern bool FMVstarted;
 extern bool renderswitch;
 extern bool EnableFMV;
 
-void DoFmvSwitch()
+void DoFmvSwitch(bool on)
 {
-	ScopedCoreThreadPause paused_core( new SysExecEvent_SaveSinglePlugin(PluginId_GS) );
-	renderswitch = !renderswitch;
-	paused_core.AllowResume();
+	if (g_Conf->GSWindow.IsToggleAspectRatioSwitch) {
+		AspectRatioType& art = g_Conf->GSWindow.AspectRatio;
+		if (!EnableFMV && !FMVstarted && art == AspectRatio_4_3 && prevAR != art)
+			prevAR = art;
+
+		if (on) {
+			if (art != AspectRatio_4_3) {
+				prevAR = art;
+				art = AspectRatio_4_3;
+			}
+		}
+		else if (!on) {
+			if (art == AspectRatio_4_3 && prevAR != AspectRatio_4_3)
+				art = prevAR;
+		}
+
```

Comment:
Fair enough , I guess. But I still don't think using ``else if ()`` at this scenario compromises readability.